File: backend/app/api/v1/unit_of_measures.py
"""
Unit of Measure API endpoints.
"""
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from typing import List

from app.database import get_db
from app.models import UnitOfMeasure
from app.api.v1.auth import get_current_user
from app.models import User

router = APIRouter(prefix="/unit-of-measures", tags=["unit-of-measures"])


@router.get("")
async def list_unit_of_measures(
    active_only: bool = True,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """List all unit of measures."""
    query = db.query(UnitOfMeasure)
    if active_only:
        query = query.filter(UnitOfMeasure.is_active == True)
    units = query.order_by(UnitOfMeasure.name).all()
    return [
        {
            "id": unit.id,
            "name": unit.name,
            "abbreviation": unit.abbreviation,
            "type": unit.type,
            "is_active": unit.is_active,
        }
        for unit in units
    ]

File: frontend/pos/src/api/unitOfMeasures.ts
/**
 * API client for unit of measures.
 */
import apiClient from './client'

export interface UnitOfMeasure {
  id: number
  name: string
  abbreviation: string
  type: string
  is_active: boolean
}

export async function listUnitOfMeasures(activeOnly: boolean = true): Promise<UnitOfMeasure[]> {
  const response = await apiClient.get('/api/v1/unit-of-measures', {
    params: { active_only: activeOnly },
  })
  return response.data
}

